cql3/statements/describe_statement.cc

```diff
@@ -0,0 +1,811 @@
+        if (with_stmt) {
+            desc.add_create_statement(get_create_statement(*uda));
+        }
+
+        return desc;
+    }));
+}
+
+description view(const data_dictionary::database& db, const sstring& ks, const sstring& name, bool with_internals) {
+    auto view = db.try_find_table(ks, name);
+    if (!view || !view->schema()->is_view()) {
+        throw exceptions::invalid_request_exception(format("Materialized view '{}' not found in keyspace '{}'", name, ks));
+    }
+
+    std::ostringstream os;
+    view->schema()->describe(db.real_database(), os, with_internals);
+    auto stmt = os.str();
+
+    return description(ks, "view", name, stmt);
+}
+
+description index(const data_dictionary::database& db, const sstring& ks, const sstring& name, bool with_internals) {
+    auto schema = db.find_indexed_table(ks, name);
+    if (!schema) {
+        throw exceptions::invalid_request_exception(format("Table for existing index '{}' not found in '{}'", name, ks));
+    }
+
+    std::ostringstream os;
+    std::optional<view_ptr> idx;
+
+    auto views = db.find_table(ks, schema->cf_name()).views();
+    for (const auto& view: views) {
+        if (is_index(db, view) && secondary_index::index_name_from_table_name(view->cf_name()) == name) {
+            idx = view;
+            break;
+        }
+    }
+
+    (*idx)->describe(db.real_database(), os, with_internals);
+    auto stmt = os.str();
+
+    return description(ks, "index", name, stmt);
+}
+
+std::vector<description> table(const data_dictionary::database& db, const sstring& ks, const sstring& name, bool with_internals) {
+    auto table = db.try_find_table(ks, name);
+    if (!table) {
+        throw exceptions::invalid_request_exception(format("Table '{}' not found in keyspace '{}'", name, ks));
+    }
+
+    std::vector<description> result;
+    std::ostringstream os;
+
+    // table
+    auto schema = table->schema();
+    schema->describe(db.real_database(), os, with_internals);
+    auto stmt = os.str();
+    result.push_back(description(ks, "table", schema->cf_name(), stmt));
+
+    // indexes
+    auto idxs = table->get_index_manager().list_indexes();
+    boost::sort(idxs, [] (const auto& a, const auto& b) {
+        return a.metadata().name() < b.metadata().name();
+    });
+
+    for (const auto& idx: idxs) {
+        result.push_back(index(db, ks, idx.metadata().name(), with_internals));
+    }
+
+    //views
+    auto views = table->views();
+    boost::sort(views, [] (const auto& a, const auto& b) {
+        return a->cf_name() < b->cf_name();
+    });
+
+    for (const auto& v: views) {
+        if(!is_index(db, v)) {
+            result.push_back(view(db, ks, v->cf_name(), with_internals));
+        }
+    }
+
+    return result;
+}
+
+std::vector<description> tables(const data_dictionary::database& db, const lw_shared_ptr<keyspace_metadata>& ks, std::optional<bool> with_internals = std::nullopt) {
+    auto&& tables = ks->tables();
```

Comment:
And here, make a copy to avoid changes to it during iteration.

Reply:
done